### PR TITLE
Auth endpoint and authenticated client

### LINKED ISSUE #46: Creating an authenticating client is kind of a mess
To create a authenticated client, you first need a client, then create a tokenGenerator and then add it back to another client.

This is kind of bad. Creating an authenticated client should be as simple as providing an APIKey to the NewClient method.

### LINKED ISSUE #47: No way to change the Auth URL
The only way to change the Auth URL is to set an environment variable. We should also have an option passed to the NewClient method.

### PR DESCRIPTION
**Pull request by johnsudaar**
Fix #46
Fix #47

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -89,6 +93,10 @@ func NewClient(cfg ClientConfig) *Client {
 		},
 	}
 
+	if len(cfg.APIToken) != 0 && c.tokenGenerator == nil {
```
Comment (EtienneM) [inline client.go]: why not `if cfg.APIToken != ""`?...

File: auth_api_token.go [hunk 2]
```diff
@@ -61,3 +54,16 @@ func (t *APITokenGenerator) GetAccessToken() (string, error) {
 	}
 	return accessToken, nil
 }
+
+func (c *Client) AuthURL() string {
+	if len(c.authEndpoint) != 0 {
```
[the remark this answers is not in the document]
Comment (johnsudaar) [reply auth_api_token.go]: Both method works. I prefer this one.

File: client.go (hunk 1 above)
Comment (johnsudaar) [reply client.go]: Both method works. I prefer this one.

File: auth_api_token.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (EtienneM) [reply auth_api_token.go]: OK I disagree with you at I find this solution less readable. But if we stick with this, why 3 lines later you don't use the same condition with `if len(os.Getenv("SCALINGO_AUTH_URL")) != 0`?

[the remark this answers is not in the document]
Comment (EtienneM) [reply auth_api_token.go]: You don't answer my point about consistency. Using one way on line 59 and another on line 62 is not something I like ^^

**Pull request merged**